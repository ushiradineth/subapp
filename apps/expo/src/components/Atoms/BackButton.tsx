import React from "react";
import { Platform, Pressable } from "react-native";
import { useRouter } from "expo-router";
import { ArrowLeft } from "lucide-react-native";

const BackButton = () => {
  const router = useRouter();

  return (
    <Pressable
      android_ripple={{ color: "gray", borderless: true, radius: 20 }}
      onPress={() => router.back()}
      className={`rounded-full ${Platform.OS === "android" ? "px-2 pb-2 pt-1" : "p-2"}`}>
      <ArrowLeft className={` ${Platform.OS === "android" && "mt-1"}`} color={"black"} />
    </Pressable>
  );
};

export default BackButton;
