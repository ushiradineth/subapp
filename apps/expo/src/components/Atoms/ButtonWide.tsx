import React from "react";
import { Text } from "react-native";
import { Button } from "tamagui";

interface Props {
  icon?: React.ReactNode;
  text: string;
  onPress: () => void;
  justifyText?: "left" | "center" | "right";
}

const ButtonWide = ({ icon, text, onPress, justifyText = "left" }: Props) => {
  return (
    <Button onPress={onPress} className={`bg-background flex h-16 w-full items-center justify-${justifyText} rounded-3xl`}>
      {icon}
      <Text className="text-[16px] font-semibold">{text}</Text>
    </Button>
  );
};

export default ButtonWide;
