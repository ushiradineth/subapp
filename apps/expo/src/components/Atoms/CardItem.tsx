import React, { type ReactNode } from "react";
import { Image, Text, YStack } from "tamagui";

interface Props {
  image: string;
  onPress: () => void;
  title: ReactNode;
  text1?: ReactNode;
}

const CardItem = ({ image, title, text1, onPress }: Props) => {
  return (
    <YStack onPress={onPress} className="w-36">
      <Image className="bg-foreground rounded-t-2xl" source={{ uri: image, height: 144, width: 144 }} alt={title?.toString()} />
      <YStack className="bg-background flex h-12 w-full justify-center rounded-b-2xl pl-2">
        <Text className="text-xs font-semibold">{title}</Text>
        {text1 && <Text className="text-xs">{text1}</Text>}
      </YStack>
    </YStack>
  );
};

export default CardItem;
