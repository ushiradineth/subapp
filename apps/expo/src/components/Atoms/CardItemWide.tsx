import React, { type ReactNode } from "react";
import { Image, Text, XStack, YStack } from "tamagui";

interface Props {
  image: string;
  onPress: () => void;
  title: ReactNode;
  text1?: ReactNode;
  text2?: ReactNode;
  text3?: ReactNode;
}

const CardItemWide = ({ title, text1, text2, text3, image, onPress }: Props) => {
  return (
    <XStack onPress={onPress} className="bg-background flex w-full items-center justify-start rounded-2xl p-2">
      <Image className="bg-foreground rounded-2xl" source={{ uri: image, height: 144, width: 144 }} alt={title?.toString()} />
      <YStack className="flex justify-center rounded-b-2xl p-2">
        <Text className="text-xs font-semibold">{title}</Text>
        {text1 && <Text className="text-xs">{text1}</Text>}
        {text2 && <Text className="text-xs">{text2}</Text>}
        {text3 && <Text className="text-xs">{text3}</Text>}
      </YStack>
    </XStack>
  );
};

export default CardItemWide;
