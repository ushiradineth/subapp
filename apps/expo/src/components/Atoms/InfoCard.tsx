import React from "react";
import { View } from "react-native";
import { Text, XStack } from "tamagui";

export default function InfoCard({ children }: { children: React.ReactNode }) {
  return (
    <XStack className="flex h-16 items-center justify-between">
      <View className="bg-background border-accent flex h-full w-full items-center justify-center rounded-3xl border">
        <Text>{children}</Text>
      </View>
    </XStack>
  );
}
