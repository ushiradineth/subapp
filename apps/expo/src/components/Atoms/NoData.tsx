import React from "react";
import { Text, View } from "react-native";
import { YStack } from "tamagui";

const NoData = ({ children, background }: { children: React.ReactNode, background?: boolean }) => {
  if (background) {
    return (
      <YStack flex={1} alignItems="center" justifyContent="center" space={4}>
        <View className="flex h-full items-center justify-center">
          <Text className="text-lg font-semibold text-gray-500">{children}</Text>
        </View>
      </YStack>
    );
  }

  return (
    <View className="flex h-full items-center justify-center">
      <Text className="text-lg font-semibold text-gray-500">{children}</Text>
    </View>
  );
};

export default NoData;
