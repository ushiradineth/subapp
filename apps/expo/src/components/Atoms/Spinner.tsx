import React from "react";
import { Spinner as Loader, YStack } from "tamagui";

interface Props {
  background?: boolean;
  color?: string;
}

export const Spinner = ({ background, color = "$accent" }: Props) => {
  if (background) {
    return (
      <YStack flex={1} alignItems="center" justifyContent="center" space={4}>
        <Loader testID="spinner-background" color={color} />
      </YStack>
    );
  }

  return <Loader testID="spinner" color={color} />;
};
