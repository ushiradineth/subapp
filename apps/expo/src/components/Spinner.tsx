import React from "react";
import { Spinner, YStack } from "tamagui";

export const SpinnerComponent = () => {
  return (
    <YStack flex={1} alignItems="center" justifyContent="center" space={4}>
      <Spinner size="large" color="$accent" />
    </YStack>
  );
};
