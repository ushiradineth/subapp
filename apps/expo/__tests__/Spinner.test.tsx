import React from "react";
import { render, screen } from "@testing-library/react-native";
import { TamaguiProvider } from "tamagui";

import { Spinner } from "../src/components/Atoms/Spinner";
import config from "../tamagui.config";

describe("Spinner", () => {
  it("renders correct children", () => {
    const { getByTestId } = render(
      <TamaguiProvider config={config}>
        <Spinner />
      </TamaguiProvider>,
    );

    expect(getByTestId("spinner")).toBeTruthy();

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("renders correct children with backgroup prop", () => {
    const { getByTestId } = render(
      <TamaguiProvider config={config}>
        <Spinner background />
      </TamaguiProvider>,
    );

    expect(getByTestId("spinner-background")).toBeTruthy();

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
