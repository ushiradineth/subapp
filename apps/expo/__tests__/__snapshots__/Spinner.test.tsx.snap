// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correct children 1`] = `
<View
  style={
    [
      {
        "alignItems": "stretch",
        "display": "flex",
        "flexBasis": "auto",
        "flexDirection": "column",
        "flexShrink": 0,
        "minHeight": 0,
        "minWidth": 0,
        "position": "relative",
      },
    ]
  }
  testID="spinner"
>
  <ActivityIndicator
    color="#5D76CC"
  />
</View>
`;

exports[`Spinner renders correct children with backgroup prop 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "flexBasis": "auto",
        "flexDirection": "column",
        "flexShrink": 0,
        "justifyContent": "center",
        "minHeight": 0,
        "minWidth": 0,
        "position": "relative",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "display": "flex",
          "flexBasis": "auto",
          "flexDirection": "column",
          "flexShrink": 0,
          "minHeight": 0,
          "minWidth": 0,
          "position": "relative",
        },
      ]
    }
    testID="spinner-background"
  >
    <ActivityIndicator
      color="#5D76CC"
    />
  </View>
</View>
`;
